nifi-commons/nifi-security-utils/src/main/java/org/apache/nifi/security/credstore/HadoopCredentialStore.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.security.credstore;
+
+import org.apache.nifi.processor.exception.ProcessException;
+
+import javax.crypto.spec.SecretKeySpec;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.KeyStore;
+import java.util.Map;
+
+public class HadoopCredentialStore {
```

Comment:
Taking a closer look, I think we would be better off using the hadoop library for this.

This module already depends on `hadoop-commons` and the API between NiFi and Atlas solidifies this dependency - although not explicitly through code, but through the constraints of how to create the keystore.

I don't think it's worth having our own version but @bbende, I'd give you the final word on this one.

Reply:
I didn't realize the Atlas bundle already had `hadoop-common` so originally I thought it was better to not add that dependency and just have our own version of this class, but if it is already there then I'm indifferent.

Reply:
I had troubles with running the tests on Windows. I managed to fix them but some other problems may occur later. So now I would also vote for not maintaining our custom solution but using the one from hadoop-common.

Reply:
Refactored to use `LocalJavaKeyStoreProvider` from `hadoop-common`.